main/src/library/Closeable.flix

```diff
@@ -0,0 +1,22 @@
+/*
+ * Copyright 2023 Paul Butcher
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+///
+/// A type class for types that can be closed.
+///
+pub class Closeable[a] {
+  pub def close(x: a): Unit \ ef
```

Comment:
I think ef should be IO

Reply:
When we discussed this [here](https://github.com/flix/flix/issues/5225#issuecomment-1369681567) you said:

> I think it has to be effect polymorphic.

Does this comment mean that you now think differently?

Reply:
Argh. Not sure. I think it depends... 😮 

I still don't fully understand what it means to close a resource.

Reply:
I'm also skeptical of the `ef` effect.

Reply:
I'm inclined to think the ultimate solution is an associated type. But `IO` would be the overapproximation.

Reply:
OK, sounds like we should go with `IO` for now and perhaps get cuter in the future. I'll make the change.